### DISCUSSION TITLE
Use metadata for recipe picture URL

### DISCUSSION THREAD

**Discussion by joerncodes** [cooklang/spec#64] [Ideas]
Hi there!

What I like about CookLang is the self-contained nature of the recipe: Everything I need is right there in the `.cook` file. This ensures things like portability and compatibility.

I am aware that adding pictures by putting a JPEG with the same name "next to" the `.cook` file is merely a convention, but I would like to suggest adding a second convention that enables providing a picture via an URL in the metadata.

It could look something like this:

```
>> source: https://www.gimmesomeoven.com/baked-potato/
>> time required: 1.5 hours
>> course: dinner
>> image: https://www.gimmesomeoven.com/wp-content/uploads/2016/03/How-To-Make-A-Baked-Potato-Recipe-2.jpg
```

By hosting my recipe images elsewhere, I could ensure that my recipes are still self-contained, while also providing pictures.

Let me know what you think!

Best,
Jörn

**Comment by dubadub** [comment]
That's a really great idea @joerncodes!

**Comment by dubadub** [comment]
Relevant proposal for spec change https://github.com/cooklang/spec/pull/105.

**Comment by dubadub** [comment]
This is now part of the main specification, closing the discussion.
